wordpress-mobile/WordPress-Android
# [Nullability Annotations to Java Classes] Replace `findViewById` with `ViewBinding` for Notifications (`relatively safe`)

Parent #18906

This PR's main focus is to replace `findViewById ` with `ViewBinding` for the [NotificationsDetailActivity.java](https://github.com/wordpress-mobile/WordPress-Android/blob/5b0d3f39c8cfe9c08b4e1547e46bc947f9be6b8f/WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java#L88) class.

Additionally, this PR is also dealing with adding any missing nullability annotations to this activity class, focusing mainly on its fields, and, any effected neighbour classes. But first, all existing warnings on this class are being resolved/suppressed so as to make it easier to deal with any missing nullability checks (with help from the IDE), especially when a `@Nullable` annotation is added.

FYI: This change is `relatively safe`, meaning that although there are compile-time changes associated with this change, and it needs testing, the changes included in this PR are very targeted and specific to one screen, and one screen only. Thus, if this screen is tested appropriately, it should be safe to merge this to `trunk`.

-----

@mkevins let me know how I did on this PR, that is, in terms of using a nullable `binding` field and the additional warning and nullability related changes I made right after, just to make this class null proof. If you also feel that this is a good way to proceed with such PRs, I'll then use this pattern across the board and follow a similar approach on every such PR.

-----

`findViewById ` -> `ViewBinding`

1. [Replace findviewbyid with viewbinding for notifications detail](https://github.com/wordpress-mobile/WordPress-Android/commit/3a376ee66bb9038f687bd124b16fde6606d16b66)

Nullability Annotation List:

1. [Add missing nullability annotations to sdk override methods](https://github.com/wordpress-mobile/WordPress-Android/commit/720bc07c414fafd9d6236672ef9ce1c18b3324c4)
2. [Replace org jetbrains with androidx non null annotations](https://github.com/wordpress-mobile/WordPress-Android/commit/36308b1ddf80bf3de7580d8020e8ee04a985f83d)
3. [Add missing nullability annotations to app override methods](https://github.com/wordpress-mobile/WordPress-Android/commit/935b7e515c1a7619cd930d313434ee3e5bbf1400)
4. [Add missing nullability annotations to note id field](https://github.com/wordpress-mobile/WordPress-Android/commit/a94295900e131c71f69b121c9beca8516d7f5f77)
5. [Add missing nullability annotations to context parameters](https://github.com/wordpress-mobile/WordPress-Android/commit/fd5b3b1860b778194d2b6785ada6c7d07ffb8ce0)
6. [Add missing nullability annotations to change listener field](https://github.com/wordpress-mobile/WordPress-Android/commit/2da778e6612d7355e15b2cbf7f41c7510f4a8b60)
7. [Add missing nullability annotations to adapter field](https://github.com/wordpress-mobile/WordPress-Android/commit/8ccc1ab9a47a597ee75d5aa3a187b016563a0303)

Warnings Resolution List:

1. [Resolve unnecessary return warning on notifications detail](https://github.com/wordpress-mobile/WordPress-Android/commit/108accbae620a9402354c50ba924d47a349ee788)

Warnings Suppression List:

1. [Suppress unchecked warning on notifications detail ](https://github.com/wordpress-mobile/WordPress-Android/commit/91babcb07d6adf914a44f8434ce36ab965d9224c)
2. [Suppress deprecation warnings on notifications detail](https://github.com/wordpress-mobile/WordPress-Android/commit/f59a4b9491653763c7a595e47cfa1d1608d20b39)

-----

## To test:

1. Quickly smoke test any notification related functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected.
8. In addition to the above smoke test, you can expand the below and follow the inner and more explicit test steps within:

<details>
    <summary>Notifications Details Screen [NotificationsDetailActivity.java]</summary>

ℹ️ This test applies to the Jetpack app.

- Go to the `Notifications` tab and tap on any notification.
- Verify that the `Notification Details` screen is shown and functioning as expected.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

9. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


-----

## UI Changes testing checklist:
